core/services/vrf/integration_v2_test.go

```diff
@@ -206,6 +214,50 @@ func newVRFCoordinatorV2Universe(t *testing.T, key ethkey.KeyV2, numConsumers in
 	require.NoError(t, err, "failed to send LINK to VRFMaliciousConsumer contract on simulated ethereum blockchain")
 	backend.Commit()
 
+	// Deploy upgradeable consumer, proxy, and proxy admin
+	upgradeableConsumerAddress, _, _, err := vrf_consumer_v2_upgradeable.DeployVRFConsumerV2Upgradeable(neil, backend)
+	require.NoError(t, err, "failed to deploy upgradeable consumer to simulated ethereum blockchain")
+	backend.Commit()
+
+	proxyAdminAddress, _, proxyAdmin, err := vrfv2_proxy_admin.DeployVRFV2ProxyAdmin(neil, backend)
+	require.NoError(t, err)
+	backend.Commit()
+
+	// provide abi-encoded initialize function call on the implementation contract
+	// so that it's called upon the proxy construction, to initialize it.
+	upgradeableAbi, err := vrf_consumer_v2_upgradeable.VRFConsumerV2UpgradeableMetaData.GetAbi()
+	require.NoError(t, err)
+	initializeCalldata, err := upgradeableAbi.Pack("initialize", coordinatorAddress, linkAddress)
+	hexified := hexutil.Encode(initializeCalldata)
+	t.Log("initialize calldata:", hexified, "coordinator:", coordinatorAddress.String(), "link:", linkAddress)
+	require.NoError(t, err)
+	proxyAddress, _, _, err := vrfv2_transparent_upgradeable_proxy.DeployVRFV2TransparentUpgradeableProxy(
+		neil, backend, upgradeableConsumerAddress, proxyAdminAddress, initializeCalldata)
+	require.NoError(t, err)
+
+	_, err = linkContract.Transfer(sergey, proxyAddress, assets.Ether(500)) // Actually, LINK
+	require.NoError(t, err)
+	backend.Commit()
+
+	implAddress, err := proxyAdmin.GetProxyImplementation(nil, proxyAddress)
+	require.NoError(t, err)
+	t.Log("impl address:", implAddress.String())
+	require.Equal(t, upgradeableConsumerAddress, implAddress)
+
+	proxiedConsumer, err := vrf_consumer_v2_upgradeable.NewVRFConsumerV2Upgradeable(
+		proxyAddress, backend)
+	require.NoError(t, err)
+
+	cAddress, err := proxiedConsumer.COORDINATOR(nil)
+	require.NoError(t, err)
+	t.Log("coordinator address in proxy to upgradeable consumer:", cAddress.String())
+	require.Equal(t, coordinatorAddress, cAddress)
+
+	lAddress, err := proxiedConsumer.LINKTOKEN(nil)
+	require.NoError(t, err)
+	t.Log("link address in proxy to upgradeable consumer:", lAddress.String())
+	require.Equal(t, linkAddress, lAddress)
```

Comment:
just to confirm my understanding, the purpose of test is to make sure the logic contract can be called with `proxyAddress` right?

Reply:
The purpose of this setup is to make sure that the state in the proxy contract makes sense, basically that the link address and the coordinator address storage slots are correctly set in the proxy contract itself.

Reply:
These are just sanity checks though, not really comprehensive tests, since if these slots don't have the correct state, nothing in the proxy tests would work.